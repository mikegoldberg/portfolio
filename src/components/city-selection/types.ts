export interface CitySelectionProps {
  city: string;
  onSubmit: (name: string) => void;
}
