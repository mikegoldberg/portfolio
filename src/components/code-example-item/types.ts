import { MouseEventHandler } from "react";

export interface CodeExampleItemProps {
  title: string;
  poster: string;
  onClick: MouseEventHandler;
  source: string;
}
