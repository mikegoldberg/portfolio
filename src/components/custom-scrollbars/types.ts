import { ReactElement } from "react";

export interface CustomScrollbarsProps {
  onScrollFrame: (position: number) => void;
  children?: ReactElement;
}
