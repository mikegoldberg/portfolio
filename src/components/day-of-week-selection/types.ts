type OnDaySelected = {
  (date: string): void;
};

export interface DayOfWeekSelectionProps {
  dates: Array<string>;
  activeDayOfWeek: string;
  onDaySelected: OnDaySelected;
}
