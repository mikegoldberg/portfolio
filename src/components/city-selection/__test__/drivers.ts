export const HEADING = "city-selection-heading";
