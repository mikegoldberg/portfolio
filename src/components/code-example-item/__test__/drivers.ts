export const POSTER = "code-example-item-poster";
export const ONCLICK = "code-example-item-onclick";
export const TITLE = "code-example-item-title";
export const SOURCE = "code-example-item-source";
