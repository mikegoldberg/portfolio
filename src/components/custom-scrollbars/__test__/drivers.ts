export const CONTENT = "custom-scrollbars-content";
