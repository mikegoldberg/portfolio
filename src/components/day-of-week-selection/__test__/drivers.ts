export const DAY_OF_WEEK = "day-of-week-selection-day-of-week";
