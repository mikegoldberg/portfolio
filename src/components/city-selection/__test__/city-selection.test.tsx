import { screen } from "@testing-library/react";
import { render } from "../../../test-utils";
import CitySelection from "..";
import * as TEST_DRIVERS from "./drivers";
import { CitySelectionProps } from "../types";

const defaultProps: CitySelectionProps = {
  city: "test city",
  onSubmit: jest.fn(),
};

describe("CitySelection", () => {
  it("renders correctly", () => {
    const { container } = render(<CitySelection {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("renders the city name", () => {
    render(<CitySelection {...defaultProps} />);
    const element = screen.getByTestId(TEST_DRIVERS.HEADING);
    expect(element).toHaveTextContent(defaultProps.city);
  });
});
