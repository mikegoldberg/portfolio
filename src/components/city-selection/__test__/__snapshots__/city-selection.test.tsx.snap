// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitySelection renders correctly 1`] = `
<div>
  <div
    class="chakra-editable css-1jijfcn"
  >
    <div
      class="css-1h2sv0j"
    >
      <h2
        class="chakra-heading css-1ixkcyz"
        data-testid="city-selection-heading"
      >
        <span
          class="chakra-editable__preview css-1gasyng"
        >
          test city
        </span>
      </h2>
      <input
        class="chakra-editable__input css-1s340xu"
        hidden=""
        value="test city"
      />
      <div
        class="css-1l4w6pd"
      >
        <button
          aria-label="Edit"
          class="chakra-button css-1gbpnky"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
